### PR TITLE
Removed metrics event

### LINKED ISSUE #13771: Remove usages of metricsEvent (old metrics context method)
We want to move away from using the old context to reduce confusion. Everywhere the `this.context.metricsEvent` function is used we want to replace with `this.context.trackEvent`. 

## Requirements
* In the file list section below, convert usages of the metricEvent context to use the new MetaMetrics context from `metametrics.new.js`/ or the trackEvent method from context.
* Using the below guide, translate the payloads of these methods to use the new payload parameters
* Delete the old context, including removing it from the root page.
* Rename `metametrics.new.js` to `metametrics.js` and update imports

## Transformation guide
Matomo event structure has two positional parameters. The first is called config and is an object. The second parameter is called overrides and is an object as well. The new context for segment is a function that takes two parameters, the first is payload and the second are options.

**The following transformations apply to the payload object, or first parameter to the method**
1. `config.eventOpts.name` -> `event`
2. `config.eventOpts.category` -> `category`
3. `config.eventOpts.action` -> `properties.action`
4. `config.customVariables` -> `properties`

**These transformations apply to the options object, or second parameter. If none of the keys exist then omit the options object**
1. `config.isOptIn` -> `isOptIn`
2. `config.eventOpts.excludeMetaMetricsId` -> `excludeMetaMetricsId`
3. `overrides.excludeMetaMetricsId` -> `excludeMetaMetricsId`
4. `config.metaMetricsId` -> `metaMetricsId`
5. `config.flushImmediately` -> `flushImmediately`

**Add the following property to the properties object for events converted**
`legacy_event: true`

### Example
```js
<PageContainerFooter
  onCancel={() => {
    setParticipateInMetaMetrics(false).then(() => {
      metricsEvent(
        {
          eventOpts: {
            category: 'Onboarding',
            action: 'Metrics Option',
            name: 'Metrics Opt Out',
          },
          isOptIn: true,
        },
        {
          excludeMetaMetricsId: true,
        },
      );
      hideModal();
    });
  }}
  cancelText={t('noThanks')}
  hideCancel={false}
  onSubmit={() => {
    setParticipateInMetaMetrics(true).then(() => {
      this.context.metricsEvent({
        eventOpts: {
          category: 'Onboarding',
          action: 'Metrics Option',
          name: 'Metrics Opt In',
        },
        isOptIn: true,
      });
      hideModal();
    });
  }}
  submitText={t('affirmAgree')}
  disabled={false}
/>
```

```js
<PageContainerFooter
  onCancel={() => {
    setParticipateInMetaMetrics(false).then(() => {
      this.context.trackEvent(
        {
            category: 'Onboarding',
            event: 'Metrics Opt Out',
            properties: {
              action: 'Metrics Option',
              legacy_event: true,
           }
        },
        {
          excludeMetaMetricsId: true,
          isOptIn: true,
        },
      );
      hideModal();
    });
  }}
  cancelText={t('noThanks')}
  hideCancel={false}
  onSubmit={() => {
    setParticipateInMetaMetrics(true).then(() => {
      metricsEvent({
        eventOpts: {
          category: 'Onboarding',
          action: 'Metrics Option',
          name: 'Metrics Opt In',
        },
        isOptIn: true,
      });
      hideModal();
    });
  }}
  submitText={t('affirmAgree')}
  disabled={false}
/>
```

## File List
- [ ] ui/pages/unlock-page/unlock-page.component.js
- [ ] ui/pages/settings/security-tab/security-tab.component.js
- [ ] ui/pages/settings/experimental-tab/experimental-tab.component.js
- [ ] ui/pages/settings/advanced-tab/advanced-tab.component.js
- [ ] ui/pages/send/send-footer/send-footer.component.js
- [ ] ui/pages/send/send-content/send-asset-row/send-asset-row.component.js
- [ ] ui/pages/onboarding-flow/metametrics/metametrics.js
- [ ] ui/pages/keychains/restore-vault.js
- [ ] ui/pages/first-time-flow/seed-phrase/reveal-seed-phrase/reveal-seed-phrase.component.js
- [ ] ui/pages/first-time-flow/seed-phrase/confirm-seed-phrase/confirm-seed-phrase.component.js
- [ ] ui/pages/first-time-flow/metametrics-opt-in/metametrics-opt-in.component.js
- [ ] ui/pages/first-time-flow/end-of-flow/end-of-flow.component.js
- [ ] ui/pages/first-time-flow/create-password/new-account/new-account.component.js
- [ ] ui/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js
- [ ] ui/pages/create-account/import-account/private-key.js
- [ ] ui/pages/create-account/import-account/json.js
- [ ] ui/pages/create-account/connect-hardware/index.js
- [ ] ui/pages/create-account/new-account.component.js
- [ ] ui/pages/confirm-transaction-base/confirm-transaction-base.component.js
- [ ] ui/pages/confirm-encryption-public-key/confirm-encryption-public-key.component.js
- [ ] ui/pages/confirm-decrypt-message/confirm-decrypt-message.component.js
- [ ] ui/pages/confirm-add-suggested-token/confirm-add-suggested-token.component.js
- [ ] ui/components/app/transaction-list-item-details/transaction-list-item-details.component.js
- [ ] ui/components/app/signature-request-original/signature-request-original.component.js
- [ ] ui/components/app/signature-request/signature-request.component.js
- [ ] ui/components/app/permission-page-container/permission-page-container.component.js
- [ ] ui/components/app/modals/metametrics-opt-in-modal/metametrics-opt-in-modal.component.js
- [ ] ui/components/app/modals/deposit-ether-modal/deposit-ether-modal.component.js
- [ ] ui/components/app/gas-customization/gas-modal-page-container/gas-modal-page-container.component.js
- [ ] ui/components/app/dropdowns/network-dropdown.js
- [ ] ui/components/app/create-new-vault/create-new-vault.js
- [ ] ui/components/app/app-header/app-header.component.js
- [ ] ui/components/app/account-menu/account-menu.component.js

### PR DESCRIPTION
**Pull request by filipsekulic**
Fixes: https://github.com/MetaMask/metamask-extension/issues/13771

1st PR as agreed: https://github.com/MetaMask/metamask-extension/pull/14042#issuecomment-1072547603

### CODE REVIEW

**Comment by brad-decker** [comment]
@filipsekulic I have a suggestion to make the review of this much easier. Lets split this into three PRS:

1. changing everything that used the metaMetrics context function to use the trackEvent context function and point all `useContext` to use `metametrics.new.js`
2. Delete `metametrics.js` (the old one) 
3. Rename `metametrics.new.js` to `metametrics.js` and update imports

Why? 

The diff on `metametrics.js` right now looks obtuse and will be a difficult review, when in fact all you are doing is renaming the file. By splitting in this way the 2 other prs will be easy to review and the majority of the work of renaming/restructuring events calls will be easier to review as well

**Comment by filipsekulic** [comment]
> @filipsekulic I have a suggestion to make the review of this much easier. Lets split this into three PRS:
> 
> 1. changing everything that used the metaMetrics context function to use the trackEvent context function and point all `useContext` to use `metametrics.new.js`
> 2. Delete `metametrics.js` (the old one)
> 3. Rename `metametrics.new.js` to `metametrics.js` and update imports
> 
> Why?
> 
> The diff on `metametrics.js` right now looks obtuse and will be a difficult review, when in fact all you are doing is renaming the file. By splitting in this way the 2 other prs will be easy to review and the majority of the work of renaming/restructuring events calls will be easier to review as well

Makes sense. I will do it like you said, so this PR will be the 1st one as you suggested.

**Comment by mirjanaKukic** [comment]
Verified by QA

File: ui/components/app/create-new-vault/create-new-vault.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default function CreateNewVault({
   const [termsChecked, setTermsChecked] = useState(false);
 
   const t = useI18nContext();
-  const metricsEvent = useContext(MetaMetricsContext);
+  const trackEvent = useContext(MetaMetricsContext);
```
Comment (brad-decker) [inline ui/components/app/create-new-vault/create-new-vault.js:25]: [REQ] in this first PR you'll need to update import statements for the `MetaMetricsContext` object to point to the the `.new` context file. Then in subsequent PRs it'll change back. This is fine, a little more work for you but ultimately easier to review and more safe with checks. Sorry for the churn!

File: ui/pages/onboarding-flow/metametrics/metametrics.js [hunk 2]
```diff
@@ -36,33 +36,43 @@ export default function OnboardingMetametrics() {
   const firstTimeSelectionMetaMetricsName =
     firstTimeFlowTypeNameMap[firstTimeFlowType];
 
-  const metricsEvent = useContext(MetaMetricsContext);
+  const trackEvent = useContext(MetaMetricsContext);
```
Comment (brad-decker) [inline ui/pages/onboarding-flow/metametrics/metametrics.js:39]: [REQ] in this first PR you'll need to update import statements for the `MetaMetricsContext` object to point to the the `.new` context file. Then in subsequent PRs it'll change back. This is fine, a little more work for you but ultimately easier to review and more safe with checks. Sorry for the churn!

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/components/app/create-new-vault/create-new-vault.js (hunk 1 above)
Comment (filipsekulic) [reply ui/components/app/create-new-vault/create-new-vault.js:25]: Fixed

File: ui/pages/onboarding-flow/metametrics/metametrics.js (hunk 2 above)
Comment (filipsekulic) [reply ui/pages/onboarding-flow/metametrics/metametrics.js:39]: Fixed

**Review by brad-decker — APPROVED** [review]

**Review by digiwand — APPROVED** [review]
🚀

**Pull request merged**